### DISCUSSION TITLE
Coordinates of the center of the pore with respect to the cartesian plane

### DISCUSSION THREAD

**Discussion by aayushi363** [PMEAL/OpenPNM#2907] [Q&A]
I am trying to find the shortest distance between to pores of opposite faces. And to specify whether the pore belongs to that face we have to compare the coordinates of the centers of each sphere (pore) relative to the minimum and maximum coordinates in each Cartesian direction across all pores in the pore network. My question is how can I find the center of the pore. Is there any property of network that returns the coordinates of the pore?

**Comment by jgostick** [comment]
All coordinates refer to the centers of the pores.  So `net['pore.coords']` is an Np-by-3 array of the x,y,z coordinates of the pore centers.

**Reply by aayushi363** [reply]
I am bit confused because here https://openpnm.org/modules/generated/openpnm.models.geometry.throat_centroid.pore_coords.html#openpnm.models.geometry.throat_centroid.pore_coords

pore.coords corresponds to "Calculate throat centroid values by averaging adjacent pore coordinates"

**Reply by jgostick** [reply]
If you're implying that the above method of calculating throat centroid is not ideal, then you're correct.  It's just an approximation.  A more correct way would be to find the 'edge' of each pore and find the midpoint between these, BUT finding edges is not as easy as you'd like.  It depends on whether the user is assuming spherical or cubic pores, for instance.

**Reply by ma-sadeghi** [reply]
@jgostick isn't that a typo in the docs?

**Reply by jgostick** [reply]
no, it uses the pore coords...otherwise we'd need 'throat.endpoints' which depends on the shape assumed for the pores.

**Reply by ma-sadeghi** [reply]
Oh right, I missed it's part of throat_centroid module, thought it was a standalone model to compute pore coords.
